### PR TITLE
Update remote flow documentation, and add a picorv32/remote tutorial.

### PR DESCRIPTION
**Pull request by WRansohoff**
The new tutorial's verbiage could use a review, but it should be fairly complete.

### CODE REVIEW

File: docs/tutorials/remote_soc.rst [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+[TODO / CR feedback: Add a screenshot here (or at the bottom)?]
+
+Installing SiliconCompiler
+--------------------------
+
+You must have a recent version of Python3 and its Pip package manager in order to install SiliconCompiler. We recommend that Windows users download Python from `https://python.org/downloads/ <https://python.org/downloads/>`_, rather than the Microsoft Store.
+
+With that, you should be able to install SiliconCompiler directly from PyPi::
+
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:20]: @WRansohoff and @gadfort : Is the expectation that all these tutorials will be completely standalone? That is - they should be able to skip our Installation instructions and walk through these instructions? If that is not the case, we should point back to our installation instructions to avoid duplicate or out-of-sync instruction.

Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst:20]: Peter mentioned this in person today. I agree that the remote-specific instructions in the PicoRV32 tutorial should be moved to the remote section.

**Comment by WRansohoff** [comment]
In response to feedback, I've moved most of the remote-specific explanations out of the PicoRV32 tutorial. I also removed some sections which won't be needed with the self-test target and default server address.

File: docs/tutorials/remote_soc.rst [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+See the :ref:`Installation <Installation>` section for information on how to install SiliconCompiler, and the :ref:`Remote Processing <Remote Processing>` section for instructions on setting up the remote workflow.
+
+[TODO / CR feedback: Add a screenshot here (or at the bottom)?]
+
+Download PicoRV32 Verilog Code
+------------------------------
+
+The heart of any digital design is its HDL code, typically written in a language such as Verilog or VHDL. High-level synthesis languages are gaining in popularity, but most of them still output their final design sources in a traditional HDL such as Verilog.
+
+PicoRV32 is an open-source implementation of a small RISC-V CPU core, the sort you might find in a low-power microcontroller. Its source code, license, and various tooling can be found `in its GitHub repository <https://github.com/YosysHQ/picorv32>`_.
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:17]: Do we want the user to use our example (since the example without RAM is the same as what's here) or download from the YosysHQ GitHub repo that's pointed here?

File: docs/tutorials/remote_soc.rst [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+Build the PicoRV32 Core using SiliconCompiler
+---------------------------------------------
+
+Before we add the complexity of a RAM macro block, let's build the core design using the open-source Skywater130 PDK. Copy the following build script into the same directory which you copied ``picorv32.v`` into::
+
+    import siliconcompiler
+
+    chip = siliconcompiler.Chip('picorv32')
+    chip.load_target('skywater130_demo')
+    chip.input('picorv32.v')
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:36]: I think you're missing:
`chip.input('picorv32.sdc')`

File: docs/tutorials/remote_soc.rst [hunk 4]
```diff
@@ -0,0 +1,209 @@
+Building Your Own SoC
+=====================
+
+This tutorial will walk you through the process of building an ASIC containing one PicoRV32 RISC-V CPU core and 2 kilobytes of SRAM, on an open-source 130nm Skywater process node.
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:4]: Explicitly add this is using remote flow:
> This tutorial will walk you through the process of building an ASIC containing one PicoRV32 RISC-V CPU core and 2 kilobytes of SRAM, on an open-source 130nm Skywater process node, **with SiliconCompiler's remote flow**.

File: docs/tutorials/remote_soc.rst [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+    import siliconcompiler
+
+    chip = siliconcompiler.Chip('picorv32')
+    chip.load_target('skywater130_demo')
+    chip.input('picorv32.v')
+    chip.set('option', 'remote', True)
+    chip.run()
+
+If you run that example as a Python script, it should take approximately 10-15 minutes to run if the servers are not too busy. We have not added a RAM macro yet, but this script will build the CPU core with I/O signals placed pseudo-randomly around the edges of the die area. Once the job finishes, you should receive a screenshot of your final design, and a report containing metrics related to the build.
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:40]: Add this to be explicit:
> and a report containing metrics related to the build, **in the `build/picorv32/<jobname>/` directory

File: docs/tutorials/remote_soc.rst (hunk 2 above)
Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst:17]: Personally, I find it helpful to have a complete example to refer to when I follow tutorials that try to teach by building a project from the ground up.

When a bunch of small code blocks are interspersed with explanations, it can sometimes be hard to follow where exactly each one is supposed to go in the partially-completed example. So I thought it might be helpful to link to known-good examples for the designs that the tutorial covers.

File: docs/tutorials/remote_soc.rst (hunk 3 above)
Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst:36]: I'm planning to make a few changes which replace the SDC files with calls to `chip.clock()`

File: docs/tutorials/remote_soc.rst (hunk 2 above)
Comment (aulihant) [reply docs/tutorials/remote_soc.rst:17]: Make sense. 
In that case, this will help make more clear this is the case, e.g, add:
> **This tutorial will walk you through downloading the design files from the source, but for your reference,** complete example designs which reflect the contents of this tutorial **are** in the public SiliconCompiler repository, [with RAM](https://github.com/siliconcompiler/siliconcompiler/tree/main/examples/picorv32_ram) and [without RAM](https://github.com/siliconcompiler/siliconcompiler/tree/main/examples/picorv32).

File: docs/tutorials/remote_soc.rst (hunk 3 above)
Comment (aulihant) [reply docs/tutorials/remote_soc.rst:36]: Do we want this to be consistent throughout all the examples? In many other places right now, it's using `chip.input('<design>.sdc'.

Comment (gadfort) [reply docs/tutorials/remote_soc.rst:36]: if the SDC is basically empty, then the .clock is probably better, otherwise we can use the SDC. The examples do not have to 100% consistent on this since we also need to show how to supply an SDC file

File: docs/tutorials/remote_soc.rst [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+In this tutorial, we'll take the first step by adding a small (2 kilobyte) SRAM block and wiring it to the CPU's memory interface. This will teach you how to import and place a hard IP block in your design.
+
+The open-source Skywater130 PDK does not currently include foundry-published memory macros. Instead, they have a set of OpenRAM configurations which are blessed by the maintainers. You can use `those configurations <https://github.com/VLSIDA/OpenRAM/tree/stable/technology/sky130>`_ to generate RAM macros from scratch if you are willing to install the `OpenRAM utility <https://github.com/VLSIDA/OpenRAM>`_, or you can `download pre-built files <https://github.com/VLSIDA/sky130_sram_macros>`_ which have been published under a permissive license. We will use the ``sky130_sram_2kbyte_1rw1r_32x512_8`` block in this example.
+
+Once you have a GDS and LEF file for your RAM macro, create a new directory called ``sram`` in same location as your PicoRV32 build files, and copy the macro files there. Then, create a Python script which describes the RAM macro in a format which can be imported by SiliconCompiler::
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:53]: Add the file name since it is referenced directly later on:
> Then, create a Python script, **called ``sky130_sram_2k.py`` in that ``sram`` directory,** which describes...

File: docs/tutorials/remote_soc.rst [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+            .dout1()
+        );
+    endmodule
+
+Finally, your core build script will need to be updated to import the new SRAM Library, and specify some extra parameters such as die size and macro placement::
+
+    import siliconcompiler
+
+    design = 'picorv32_top'
+    die_width = 1000
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:168]: Later is referenced as `die_w`

File: docs/tutorials/remote_soc.rst [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+    die_height = 1000
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:169]: Later is referenced as `die_h`

File: docs/tutorials/remote_soc.rst (hunk 2 above)
Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst:17]: I ended up rewriting the paragraph a bit:

> We will walk through the process of downloading the design files and writing a build script, but for your reference, you can find complete example designs which reflect the contents of this tutorial in the public SiliconCompiler repository. The first part of the tutorial will cover building the CPU core [without RAM], and the second part will describe how to [add an SRAM block].

File: docs/tutorials/remote_soc.rst (hunk 7 above)
Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst:168]: Good catch, thanks - I think `width`/`height` are probably better for a tutorial since they're more descriptive than `w`/`h`.

File: docs/tutorials/remote_soc.rst [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+
+    # Place macro instance.
+    chip.set('constraint', 'component', 'sram', 'placement', (500.0, 250.0, 0.0))
+    chip.set('constraint', 'component', 'sram', 'rotation', 270)
+
+    # Build on a remote server.
+    chip.set('option', 'remote', True)
+    chip.run()
+
+With all of that done, your top-level build script should take about 20 minutes to run on the cloud servers if they are not too busy, with most of that time spent in the routing task. As with the previous designs, you should see periodic updates on its progress, and you should receive a screenshot and metrics summary once the job is complete.
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:202]: I think it would be helpful at this point, to provide what the directory tree looks like, for the user, as a sanity check to make sure all the files got copied over okay and named what the python script is expecting. 

This was my directory tree when I got it to work:
```
<rundir>
├── sram
│ ├── sky130_sram_2k.py
│ ├── sky130_sram_2kbyte_1rw1r_32x512_8.gds
│ └── sky130_sram_2kbyte_1rw1r_32x512_8.lef
├── picorv32.py
├── picorv32.sdc
├── picorv32.v
├── picorv32_top.py
├── picorv32_top.v
└── sky130_sram_2k.bb.v
```

File: docs/tutorials/remote_soc.rst [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+        # Create library Chip object.
+        lib = siliconcompiler.Library(chip, design)
+        lib.set('output', stackup, 'gds', f'sram/sky130_sram_2kbyte_1rw1r_32x512_8.gds')
+        lib.set('output', stackup, 'lef', f'sram/sky130_sram_2kbyte_1rw1r_32x512_8.lef')
+        # Set the 'copy' field to True, to pull these files
+        # into the build directory during the 'import' task.
+        lib.set('output', stackup, 'gds', True, field='copy')
+        lib.set('output', stackup, 'lef', True, field='copy')
+
+    return lib
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:71]: I think there is an indent missing here

File: docs/tutorials/remote_soc.rst (hunk 9 above)
Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst:202]: Good idea. Maybe we should also have them put the SRAM blackbox file in the `sram/` directory?

**Comment by aulihant** [comment]
I followed the picorv32 tutorial all the way through the instructions and copied/pasted the files and ran. The good news is that with the minor notes I suggested for the copy/pasted files, it runs through!

File: docs/tutorials/remote_soc.rst (hunk 10 above)
Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst:71]: You are correct. Sorry that I didn't actually follow this tutorial end-to-end before submitting it for review.

File: docs/tutorials/remote_soc.rst (hunk 9 above)
Comment (aulihant) [reply docs/tutorials/remote_soc.rst:202]: It would be nicer, but I think that it's okay as is, for now, since this is working and matches with both the code and instructional text.
Maybe a nice-to-have for the next rev, and easier to track/follow encapsulated in its own PR.

Comment (WRansohoff) [reply docs/tutorials/remote_soc.rst]: Oh, whoops - I made the change in the last commit. It was quick because there aren't that many places where a `.bb.v` file extension is used.

File: docs/index.rst [hunk 11]
```diff
@@ -37,6 +37,7 @@ Welcome to SiliconCompiler's documentation!
    :numbered:
 
    tutorials/zerosoc
+   tutorials/remote_soc
```
Comment (gadfort) [inline docs/index.rst:40]: Lets move this below the frontend tutorials

File: docs/tutorials/remote_soc.rst [hunk 12]
```diff
@@ -0,0 +1,220 @@
+Building Your Own SoC
+=====================
+
+This tutorial will walk you through the process of building an ASIC containing one PicoRV32 RISC-V CPU core and 2 kilobytes of SRAM, on an open-source 130nm Skywater process node, with SiliconCompiler's remote workflow.
+
+We will walk through the process of downloading the design files and writing a build script, but for your reference, you can find complete example designs which reflect the contents of this tutorial in the public SiliconCompiler repository. The first part of the tutorial will cover building the CPU core `without RAM <https://github.com/siliconcompiler/siliconcompiler/tree/main/examples/picorv32>`_, and the second part will describe how to `add an SRAM block <https://github.com/siliconcompiler/siliconcompiler/tree/main/examples/picorv32>`_.
+
+See the :ref:`Installation <Installation>` section for information on how to install SiliconCompiler, and the :ref:`Remote Processing <Remote Processing>` section for instructions on setting up the remote workflow.
+
+[TODO / CR feedback: Add a screenshot here (or at the bottom)?]
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:10]: I think you can add a screenshot here if you think it looks good and otherwise you can do it below as well

File: docs/tutorials/remote_soc.rst [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+
+Download PicoRV32 Verilog Code
+------------------------------
+
+The heart of any digital design is its HDL code, typically written in a language such as Verilog or VHDL. High-level synthesis languages are gaining in popularity, but most of them still output their final design sources in a traditional HDL such as Verilog.
+
+PicoRV32 is an open-source implementation of a small RISC-V CPU core, the sort you might find in a low-power microcontroller. Its source code, license, and various tooling can be found `in its GitHub repository <https://github.com/YosysHQ/picorv32>`_.
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:17]: If we have the file, I think we should reference the source like you have, but then provide a direct link to the file in our repo

File: docs/tutorials/remote_soc.rst [hunk 14]
```diff
@@ -0,0 +1,220 @@
...
+
+The repository contains many files, but the core CPU design is located in a single file called ``picorv32.v`` at the root of the repository.
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:19]: just refer to our file

File: docs/tutorials/remote_soc.rst [hunk 15]
```diff
@@ -0,0 +1,220 @@
...
+
+Create a new directory for this project, and copy the ``picorv32.v`` file into it.
+
+Build the PicoRV32 Core using SiliconCompiler
+---------------------------------------------
+
+Before we add the complexity of a RAM macro block, let's build the core design using the open-source Skywater130 PDK. Copy the following build script into the same directory which you copied ``picorv32.v`` into::
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:26]: make a link to the skywater130 demo :ref:`skywater130-demo`

File: docs/tutorials/remote_soc.rst [hunk 16]
```diff
@@ -0,0 +1,220 @@
...
+
+    chip = siliconcompiler.Chip('picorv32')
+    chip.load_target('skywater130_demo')
+    chip.input('picorv32.v')
+    chip.set('option', 'remote', True)
+    chip.run()
+
+If you run that example as a Python script, it should take approximately 10-15 minutes to run if the servers are not too busy. We have not added a RAM macro yet, but this script will build the CPU core with I/O signals placed pseudo-randomly around the edges of the die area. Once the job finishes, you should receive a screenshot of your final design, and a report containing metrics related to the build in the ``build/picorv32/job0/`` directory.
+
+For the full GDS-II results and intermediate build artifacts, you can install the EDA tools on your local system, and run the same Python build script with the :keypath:`option, remote` parameter set to ``False``. We are not returning the full results during this early beta period because we want to minimize bandwidth, and we believe that our public beta is currently best suited for rapid prototyping and design exploration.
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:38]: We should make the "We are not returning the full results during this early beta period because we want to minimize bandwidth, and we believe that our public beta is currently best suited for rapid prototyping and design exploration." a .. note:: so people see it clearly (also add that to the remote processing).

File: docs/tutorials/remote_soc.rst [hunk 17]
```diff
@@ -0,0 +1,220 @@
...
+
+A CPU core is not very useful without any memory. Indeed, a real system-on-chip would need quite a few supporting IP blocks to be useful in the real world. At the very least, you would want a SPI interface for communicating with external non-volatile memory, a UART to get data in and out of the core, a debugging interface, and a small on-die cache.
+
+In this tutorial, we'll take the first step by adding a small (2 kilobyte) SRAM block and wiring it to the CPU's memory interface. This will teach you how to import and place a hard IP block in your design.
+
+The open-source Skywater130 PDK does not currently include foundry-published memory macros. Instead, they have a set of OpenRAM configurations which are blessed by the maintainers. You can use `those configurations <https://github.com/VLSIDA/OpenRAM/tree/stable/technology/sky130>`_ to generate RAM macros from scratch if you are willing to install the `OpenRAM utility <https://github.com/VLSIDA/OpenRAM>`_, or you can `download pre-built files <https://github.com/VLSIDA/sky130_sram_macros>`_ which have been published under a permissive license. We will use the ``sky130_sram_2kbyte_1rw1r_32x512_8`` block in this example.
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:47]: I think you can drop the "which have been published under a permissive license"

Comment (gadfort) [inline docs/tutorials/remote_soc.rst:47]: Add direct link to sky130_sram_2kbyte_1rw1r_32x512_8, if possible

File: docs/tutorials/remote_soc.rst [hunk 18]
```diff
@@ -0,0 +1,220 @@
...
+    │ ├── sky130_sram_2kbyte_1rw1r_32x512_8.gds
+    │ └── sky130_sram_2kbyte_1rw1r_32x512_8.lef
+    ├── picorv32.py
+    ├── picorv32.v
+    ├── picorv32_top.py
+    └── picorv32_top.v
+
+Your ``picorv32_top.py`` build script should take about 20 minutes to run on the cloud servers if they are not too busy, with most of that time spent in the routing task. As with the previous designs, you should see periodic updates on its progress, and you should receive a screenshot and metrics summary once the job is complete.
+
+[TODO / CR feedback: Add a screenshot here (or near the top)?]
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:215]: Definitely add Screenshot here.

File: docs/tutorials/remote_soc.rst [hunk 19]
```diff
@@ -0,0 +1,220 @@
...
+    import siliconcompiler
+
+    chip = siliconcompiler.Chip('picorv32')
+    chip.load_target('skywater130_demo')
+    chip.input('picorv32.v')
+    chip.set('option', 'remote', True)
+    chip.run()
+
+If you run that example as a Python script, it should take approximately 10-15 minutes to run if the servers are not too busy. We have not added a RAM macro yet, but this script will build the CPU core with I/O signals placed pseudo-randomly around the edges of the die area. Once the job finishes, you should receive a screenshot of your final design, and a report containing metrics related to the build in the ``build/picorv32/job0/`` directory.
+
```
Comment (gadfort) [inline docs/tutorials/remote_soc.rst:37]: Add Screenshot here.

File: docs/user_guide/remote_processing.rst [hunk 20]
```diff
@@ -7,98 +7,51 @@ The SiliconCompiler project supports a remote processing model that leverages th
  #. Warehouse scale elastic compute.
  #. NDA encumbered IPs, PDKs, and EDA tools.
 
-For command line compilation, remote processing is turned on with the '-remote' option. Results from a remote compilation should be identical to results from a local compilation.
+Note that our public beta currently only supports open-source tools and PDKs. You can access the public beta without a signup or login, and it is designed to delete your data after your jobs finish, but it is not intended to process proprietary or restricted intellectual property! Please `review our terms of service <https://www.siliconcompiler.com/terms-of-service>`_, and do not submit IP which you are not allowed to distribute.
```
Comment (gadfort) [inline docs/user_guide/remote_processing.rst:10]: Maybe put this in a .. note:: so it pops out

File: docs/user_guide/remote_processing.rst [hunk 21]
```diff
@@ -7,98 +7,51 @@ The SiliconCompiler project supports a remote processing model that leverages th
...
+
+Even though our publicly-available servers only support open-source IP and tools, the remote API is capable of supporting any SiliconCompiler modules which the server operators wish to install, and we provide a minimal example development server which can be used as a starting point for custom server implementations. You can also find descriptions of the core remote API calls in the :ref:`remote API <Server API>` section.
+
```
Comment (gadfort) [inline docs/user_guide/remote_processing.rst:13]: Add comment about only returning the summary.

File: docs/user_guide/remote_processing.rst [hunk 22]
```diff
@@ -7,98 +7,51 @@ The SiliconCompiler project supports a remote processing model that leverages th
...
+Quickstart Guide
+----------------
+
+To perform a minimal self-test using the remote flow, you can run the following commands::
+
+    pip install siliconcompiler
+    sc -target asic_demo -remote
+
+The job should only take a few minutes to run if the servers aren't too busy, and once it completes, you should receive a PDF file containing a screenshot and metrics for the build results. The self-test design is a simple 8-bit counter, so your results should look like a small block of digital logic.
+
```
Comment (gadfort) [inline docs/user_guide/remote_processing.rst:23]: Add the screenshot here. That way you don't really need the "so your results should look like a small block of digital logic." part

File: docs/user_guide/remote_processing.rst [hunk 23]
```diff
@@ -7,98 +7,51 @@ The SiliconCompiler project supports a remote processing model that leverages th
...
+Configuring a Different Remote Server
+-------------------------------------
+
+You can run the ``sc-configure`` command to configure your SiliconCompiler installation with a custom remote endpoint, if you have one. If your remote server requires authentication, you can run the utility with no arguments and fill in the address, username, and password fields that it prompts you for. If your server does not require authentication, you can simply pass its address in as a command-line argument:
```
Comment (gadfort) [inline docs/user_guide/remote_processing.rst:27]: :ref:`sc-configure`

File: docs/user_guide/remote_processing.rst [hunk 24]
```diff
@@ -7,98 +7,51 @@ The SiliconCompiler project supports a remote processing model that leverages th
...
-   * - ``eda, ..., variable, ...``
-     - None
+Remote processing is also supported from the Python interface when the :keypath:`option, remote` parameter is set to 'True'. To get a feel for the Python interface, you can try running some of our `example designs <https://github.com/siliconcompiler/siliconcompiler/tree/main/examples/>`_ with this flag enabled::
+
+  chip.set('option', 'remote', True)
+
+Troubleshooting
+---------------
+
+Our public beta servers do not prune or pre-process Schema parameters, in order to make the remote processing environment as close to a local environment as possible. The jobs will be run in isolated environments with limited communication interfaces, however, so they jobs may not be able to perform network calls or meaningful filesystem access when they are run on our cloud beta servers.
```
Comment (gadfort) [inline docs/user_guide/remote_processing.rst:53]: "however, so they jobs may not be able to perform network calls or meaningful filesystem access when they are run on our cloud beta servers." this part is a little hard to follow, I think it's just missing some text

File: docs/user_guide/remote_processing.rst [hunk 25]
```diff
@@ -7,98 +7,51 @@ The SiliconCompiler project supports a remote processing model that leverages th
...
+
+Any changes that you make to SiliconCompiler's built-in tool setup scripts on your local machine will not be reflected in jobs which are run on a remote server. Likewise, any changes that you make to the built-in open-source PDKs and standard cell libraries will not be sent to the remote servers.
```
Comment (gadfort) [inline docs/user_guide/remote_processing.rst:55]: Maybe add a link to the contributions page and encourage them to contribute the changes

**Review by gadfort — CHANGES_REQUESTED** [review]
Overall looks good, just some changes needed and maybe a few things we should chat about tomorrow

File: docs/tutorials/remote_soc.rst [hunk 26]
```diff
@@ -0,0 +1,222 @@
...
+In this tutorial, we'll take the first step by adding a small (2 kilobyte) SRAM block and wiring it to the CPU's memory interface. This will teach you how to import and place a hard IP block in your design.
+
+The open-source Skywater130 PDK does not currently include foundry-published memory macros. Instead, they have a set of OpenRAM configurations which are blessed by the maintainers. You can use `those configurations <https://github.com/VLSIDA/OpenRAM/tree/stable/technology/sky130>`_ to generate RAM macros from scratch if you are willing to install the `OpenRAM utility <https://github.com/VLSIDA/OpenRAM>`_, or you can `download pre-built files <https://github.com/VLSIDA/sky130_sram_macros>`_. We will use the `sky130_sram_2kbyte_1rw1r_32x512_8 <https://github.com/VLSIDA/sky130_sram_macros/tree/main/sky130_sram_2kbyte_1rw1r_32x512_8>` block in `this example.
```
Comment (aulihant) [inline docs/tutorials/remote_soc.rst:49]: What do you think about adding this? Might be helpful:
> Here is a shortcut to grab the lef and gds files if you find it helpful:
```
curl https://raw.githubusercontent.com/VLSIDA/sky130_sram_macros/main/sky130_sram_2kbyte_1rw1r_32x512_8/sky130_sram_2kbyte_1rw1r_32x512_8.gds > sky130_sram_2kbyte_1rw1r_32x512_8.gds
curl https://raw.githubusercontent.com/VLSIDA/sky130_sram_macros/main/sky130_sram_2kbyte_1rw1r_32x512_8/sky130_sram_2kbyte_1rw1r_32x512_8.lef > sky130_sram_2kbyte_1rw1r_32x512_8.lef

**Comment by WRansohoff** [comment]
Okay, I think that I've applied all of our online and offline comments. In particular, I added screenshots where requested, marked the tutorial code blocks with captions, and did a quick visual review of the generated HTML pages.

**Comment by gadfort** [comment]
I'm getting a: /tutorials/remote_soc.rst:24: WARNING: undefined label: 'skywater130-demo'

**Comment by WRansohoff** [comment]
This PR was split into #1365 and #1361, which are both merged now.
